ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/metrics/DeltaFilesMetricReporter.java

```diff
@@ -115,41 +117,45 @@ public static DeltaFilesMetricReporter getInstance() {
     return InstanceHolder.instance;
   }
 
-  public static synchronized void init(HiveConf conf){
+  public static synchronized void init(HiveConf conf) {
     getInstance().configure(conf);
   }
 
   public void submit(TezCounters counters) {
-    updateMetrics(NUM_OBSOLETE_DELTAS,
-        obsoleteDeltaCache, obsoleteDeltaTopN, obsoleteDeltasThreshold, counters);
-    updateMetrics(NUM_DELTAS,
-        deltaCache, deltaTopN, deltasThreshold, counters);
-    updateMetrics(NUM_SMALL_DELTAS,
-        smallDeltaCache, smallDeltaTopN, deltasThreshold, counters);
+    if(acidMetricsExtEnabled) {
```

Comment:
It makes more sense to add this check to org.apache.hadoop.hive.ql.exec.tez.TezTask#execute instead, so all the checks are in one place:
```
          if (HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_SERVER2_METRICS_ENABLED)) {
            DeltaFilesMetricReporter.getInstance().submit(dagCounters);
          }
```

Reply:
yeah.. So first issue is that this metrics is collected in HS2 but the conf (METASTORE_ACIDMETRICS_EXT_ON) for the metrics collection is defined in the MetastoreConf, so I wanted to minimize the exposure of that. 
Second is in the end I put it here, because it seamed to make the class more resilient. Here is my reasoning: This is a singleton class, so you can access it basically anywhere, but you need to call the init method on it so that it works properly. The acidMetricsExtEnabled flag is indeed set in the init (configure), so if it was not called it will skip this code part which otherwise would throw a NPE.
But I'm open to a better solution :)

Reply:
Ok, you convinced me